CIP-????/README.md

```diff
@@ -0,0 +1,259 @@
+        window.cardano[args.api.name] = api;
+        console.log(`[info]: injected api of ${args.api.name} into window.cardano`);
+    });
+</script>
+```
+
+#### Wallet App
+
+The wallet app is a standard ionic react app built by the ionic cli:
+
+```zsh
+ionic start WalletApp blank --type=react
+cd WalletApp
+npm i bugout
+```
+
+The following lines of code were added to the index.tsx file:
+
+```js
+const bugout = new Bugout(
+  "<HASH provided by the dAPP>", {
+    announce: [
+      'udp://tracker.opentrackr.org:1337/announce', 
+      'udp://open.tracker.cl:1337/announce', 
+      'udp://opentracker.i2p.rocks:6969/announce', 
+      'https://opentracker.i2p.rocks:443/announce',
+      'wss://tracker.files.fm:7073/announce',
+      'wss://spacetradersapi-chatbox.herokuapp.com:443/announce',
+      'ws://tracker.files.fm:7072/announce'
+    ]
+  });
+
+bugout.on("server", function() {
+  console.log("[info]: connected to server")
+  bugout.rpc("<HASH provided by the dAPP>", "api", {"api": {
+    version: "1.0.3",
+    name: 'boostwallet',
+    methods: ["getRewardAddresses"]
+  }});
+});
+
+bugout.register("getRewardAddresses", (address:string, args:any, callback:Function) => {
+    callback(["e1820506cb0ce54ae755b2512b6cf31856d7265e8792cb86afc94e0872"]);
+});
+```
+
+This example has a few restrictions:
+
+1. bugout is currently not compatible with Webpack 5, so polyfills are not automatically included and a react-scripts eject is needed to add them to the webpack.config.js file
+
+2. bugout does not directly provide type declarations. There are some declarations within a [PR](https://github.com/chr15m/bugout/pull/45), but they need to be adjusted (a few parameters are not mandatory) and added to a bugout.d.ts file.
+ 
+### User Flow
+
+```mermaid
+sequenceDiagram
+    dApp-->>Wallet: Share address using Deeplink /Universal Link / QR
+    Wallet-->>Tracker List: Accept connection and start quering using the address
+    Wallet-->>dApp: Once connected register RPC functions and share the API
+    dApp-->>dApp: Inject the API to window.cardano[walletName]
+    dApp-->>Wallet: Call RPC functions (e.g. getRewardAddresses)
+    Wallet-->>Wallet: If needed bring app to foregrund (e.g. request signing)
+    Wallet-->>dApp: Send response
+```
+
+### Security Aspects
+
+We decided to spawn the server within the dApp to force the user to manually scan a QR code (using a wallet app) or accept an "Open with `<WalletAppName>`" ui dialog (in case of Universal Links or Deeplinks). This prevents the user from connecting the wallet to an unwanted dApp. Additionally we need to add  a few security checks to prevent a misusage of this method.
+
+- The wallet app needs to verifiy the origin (address) of the RPC call
+- dApps should ask the user for permission to inject the wallet names into the window.cardano object to prevent XSS attack (Maybe using a graphical representation of the wallet app address e.g. blockies)
+
+## Rationale
```

Comment:
To me this rationale section is a bit light, id like to see answers too:
- Were there any alternate designs considered? Why did the current design succeed? 
  - maybe [#395 comment](https://github.com/cardano-foundation/CIPs/pull/395#issuecomment-1346314526) applies.
- What are the benefits and drawbacks of this approach when compared to the connection via CIP-30?

Reply:
As I remember @fabianbormann you/we talked about some alternative design considerations in the CIP meetings around the time this was originally presented... so perhaps you could recall & include an inventory of those that came up?